### PR TITLE
Handle NaN in toDouble.

### PR DESCRIPTION
**Pull request by krm1312**
#1482

### CODE REVIEW

**Review by dengliming — APPROVED** [review]
Thanks. LGTM

**Comment by dengliming** [comment]
Seems we should merge to `main ` branch? @mp911de

**Comment by mp911de** [comment]
There's no test attached. Can you @dengliming create one during the merge? We should port the change (via `git cherry-pick`) to `5.3.x` and `main`. 

FTR, edit the commit message to match the Lettuce commit message format (here are two commits for a PR merge: 5810f96eb6537bd88d5a59de2f202f7ac75c0ab0 and b240c5891e901183b5b762383272f187813999f6).

**Comment by dengliming** [comment]
Can we just add a simple test for method `toDouble`? I haven't find a command returning like `nan` this. This case only happen in redis-search (redis module). WDYT? @mp911de

**Comment by mp911de** [comment]
That's perfectly fine to go with a unit test as that's the simple-most approach.

**Comment by krm1312** [comment]
Thanks @dengliming @mp911de .  Let me know if you need me to clean anything up or otherwise change anything.  I put PR to 6.0.x because that's the release branch I was hoping it would get in.  Further back is of course fine w/ me as well.

**Comment by dengliming** [comment]
@krm1312  It would be better If you can also add a unit test for `LettuceStrings.toDouble`

File: src/test/java/io/lettuce/core/internal/LettuceStringsTests.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package io.lettuce.core.internal;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+
```
Comment (dengliming) [inline src/test/java/io/lettuce/core/internal/LettuceStringsTests.java:21]: Add yourself to this file as `@author`

**Comment by dengliming** [comment]
@krm1312 Thanks for your contribution. That's merged.

**Comment by dengliming** [comment]
@mp911de `LettuceStrings` This file is in conflict with the version `5.3.x` and the package name is also different. Do I need to keep it?

**Comment by mp911de** [comment]
Great merge, @dengliming 👍

Fo 5.3.x, we need to apply the fix manually, since we can't backport the commit.

**Comment by dengliming** [comment]
@mp911de Got it. That‘s backported now.

**Pull request closed without merging**